### PR TITLE
Adds `pandas.Series.str` methods

### PR DESCRIPTION
**Pull request by adimyth**
Added validation for the following set of string methods -
* count
* endswith
* find
* findall
* get
* index
* join
* len
* ljust
* lower
* lstrip
* match
* pad
* repeat
* rfind
* rindex
* rjust
* rstrip
* slice
* slice_replace
* split
* rsplit
* startswith
* strip
* translate
* wrap
* zfill
* get_dummies

### CODE REVIEW

File: pandas-stubs/core/strings.pyi [hunk 1]
```diff
@@ -1,6 +1,46 @@
...
 from .series import Series
 
 class StringMethods:
     def contains(
         self, pat: str, case: bool = ..., flags: int = ..., na: float = ..., regex: bool = ...
     ) -> Series[bool]: ...
+    def count(self, pat: str, flags: int = ...) -> Series[int]: ...
+    def endswith(self, pat: str, na: float = ...) -> Series[bool]: ...
+    def find(self, sub: str, start: int, end: int) -> Series[int]: ...
+    def findall(self, sub: str, flags: int = ...) -> Series[List[str]]: ...
```
Comment (thomkeh) [inline pandas-stubs/core/strings.pyi:12]:
```suggestion
    def findall(self, sub: str, flags: int = ...) -> Series[list]: ...
```

File: pandas-stubs/core/strings.pyi [hunk 2]
```diff
@@ -1,6 +1,46 @@
...
+    def get(i: int) -> Series: ...
```
Comment (thomkeh) [inline pandas-stubs/core/strings.pyi:13]:
```suggestion
    def get(self, i: int) -> Series: ...
```

File: pandas-stubs/core/strings.pyi [hunk 3]
```diff
@@ -1,6 +1,46 @@
...
+    def rstrip(self, to_strip: Optional[str] = ...) -> Series: ...
+    def slice(self, start: Optional[int], stop: Optional[int], step: Optional[int]) -> Series: ...
+    def slice_replace(
+        self, start: Optional[int], stop: Optional[int], repl: Optional[str]
+    ) -> Series: ...
+    def split(self, pat: Optional[str], n: int = ..., expand: bool = ...) -> Series: ...
+    def rsplit(self, pat: Optional[str], n: int = ..., expand: bool = ...) -> Series: ...
+    def startswith(self, pat: str, na: float = ...) -> Series: ...
+    def strip(self, to_strip: Optional[str] = ...) -> Series: ...
+    def translate(self, start, table: dict) -> Series: ...
```
Comment (thomkeh) [inline pandas-stubs/core/strings.pyi:35]: This doesn't seem to have a `start` argument?
```suggestion
    def translate(self, table: dict) -> Series: ...
```

**Review by thomkeh — CHANGES_REQUESTED** [review]
Thanks for the PR! There are a few small things.

**Comment by thomkeh** [comment]
I'll just merge my changes into your branch. Hope that's ok.

**Review by thomkeh — APPROVED** [review]

**Comment by thomkeh** [comment]
I hope I didn't break anything. Feel free to send another PR if you see problems with this.

**Comment by adimyth** [comment]
Cool @thomkeh Will review later tonight. Cheers!

**Pull request merged**